Python/pystate.c

```diff
@@ -622,9 +622,7 @@ init_interpreter(PyInterpreterState *interp,
     }
     interp->sys_profile_initialized = false;
     interp->sys_trace_initialized = false;
-    interp->optimizer = &_PyOptimizer_Default;
-    interp->optimizer_backedge_threshold = _PyOptimizer_Default.backedge_threshold;
-    interp->optimizer_resume_threshold = _PyOptimizer_Default.backedge_threshold;
+    _Py_SetOptimizer(interp, NULL);
```

Comment:
Hm, how about?

```suggestion
    _PyOptimizerObject *old = _Py_SetOptimizer(interp, NULL);
    assert(old == NULL);
```

...or:

```suggestion
    (void)_Py_SetOptimizer(interp, NULL);
```

Reply:
I'll go with your second suggestion. Avoiding compiler warnings for variables only used in asserts is a pain.